src/track/beatmap.cpp

```diff
@@ -614,6 +623,22 @@ void BeatMap::scaleTriple() {
     }
 }
 
+void BeatMap::scaleQuadruple() {
+    Beat prevBeat = m_beats.first();
+    // Skip the first beat to preserve the first beat in a measure
+    BeatList::iterator it = m_beats.begin() + 1;
+    for (; it != m_beats.end(); ++it) {
+        // Need to not accrue fractional frames.
+        int distance = it->frame_position() - prevBeat.frame_position();
+        Beat beat;
+        for (int i=1; i<=3, i++) {
```

Comment:
The spaces are optional, but the initialization, the comparison, and the increment need to be separated with semicolons (not commas). Looks good now!